File: rename.rb
#!/usr/bin/env ruby
require 'exifr'

files = Dir[File.join(ARGV[0], "/**/*")].select { |name| name.downcase =~ /\.(jpeg|jpg|JPG)/ }

rejected_files = []

files.each do |filename|
  date_time = EXIFR::JPEG.new(filename).date_time

  if date_time
    formatted_date = date_time.to_datetime.iso8601
    pieces = filename.split("/")
    new_filename = (pieces[0...-1] << "#{formatted_date}.jpg").join('/')
     `mv "#{filename}" "#{new_filename}"`
  else
    rejected_files << filename
  end
end

if rejected_files.size == 0
  puts "Cool! all renamed."
else
  puts "Sorry, we could not rename these files"
  puts rejected_files.inspect
end

File: rename.js
const glob = require('glob');
const fs = require('fs');
const moment = require('moment');

const directory = process.argv[2];

glob(`${directory}/**/*.@(jpeg|jpg|JPG)`, (err, files) => {
  if (err) { throw err; }
  files.forEach((filename) => {
    fs.stat(filename, (err, stats) => {
      const newName = moment(stats.birthtime).format('YYYY-MM-DDTHHmmss');
      const ext = filename.split('.').pop();
      let path = filename.split('/');
      path.pop();

      const newPath = `${path.join('/')}/${newName}.${ext}`.replace(/\:/g, '-');
      fs.rename(filename, newPath, (err) => {
        if (err) { throw err; }
        console.log('File renamed.');
      });
    });
  });
});
